lib/debug/frame_info.rb

```diff
@@ -0,0 +1,97 @@
+module DEBUGGER__
+  class FrameInfo
+    HOME = ENV['HOME'] ? (ENV['HOME'] + '/') : nil
+
+    def path
+      location.path
+    end
+
+    def pretty_path
+      use_short_path = CONFIG[:use_short_path]
+
+      case
+      when use_short_path && path.start_with?(dir = RbConfig::CONFIG["rubylibdir"] + '/')
+        path.sub(dir, '$(rubylibdir)/')
+      when use_short_path && Gem.path.any? do |gp|
+          path.start_with?(dir = gp + '/gems/')
+        end
+        path.sub(dir, '$(Gem)/')
+      when HOME && path.start_with?(HOME)
+        path.sub(HOME, '~/')
+      else
+        path
+      end
+    end
+
+    def pretty_location
+      " at #{pretty_path}:#{location.lineno}"
+    end
+
+    def parameters_info vars
+      vars.map{|var|
+        begin
+          "#{var}=#{short_inspect(binding.local_variable_get(var))}"
+        rescue NameError, TypeError
+          nil
+        end
+      }.compact.join(', ')
+    end
+
+    def klass_sig
+      if self.class == get_singleton_class(self.self)
+        "#{self.self}."
+      else
+        "#{self.class}#"
+      end
+    end
+
+    SHORT_INSPECT_LENGTH = 40
+
+    def short_inspect obj
+      str = obj.inspect
+      if str.length > SHORT_INSPECT_LENGTH
+        str[0...SHORT_INSPECT_LENGTH] + '...'
+      else
+        str
+      end
+    end
+
+    def get_singleton_class obj
+      obj.singleton_class # TODO: don't use it
+    rescue TypeError
+      nil
+    end
+
+    def to_string
```

Comment:
`to_string` is not common name, I guess. Maybe `format` or something is better because it can accept configuration.

Reply:
I am thinking `to_s` because it's essentially the string form of the frame, wdyt? Another option I have is `to_client_output`.